src/components/username-input.js

```diff
@@ -1,58 +1,55 @@
 import React from 'react';
-import autoBind from 'auto-bind';
 import PropTypes from 'prop-types';
 import { Form, Input } from 'antd';
-import { withUser } from './user-context.js';
-import { withTranslation } from 'react-i18next';
-import { userProps, translationProps, formItemLayoutShape } from '../ui/default-prop-types.js';
+import { useUser } from './user-context.js';
+import { useTranslation } from 'react-i18next';
+import { formItemLayoutShape } from '../ui/default-prop-types.js';
 
 const FormItem = Form.Item;
 
-class UsernameInput extends React.Component {
-  constructor(props) {
-    super(props);
-    autoBind(this);
-  }
+function UsernameInput({ forbiddenUsernames, formItemLayout }) {
+  const user = useUser();
+  const { t } = useTranslation('usernameInput');
 
-  render() {
-    const { forbiddenUsernames, formItemLayout, user, t } = this.props;
-
-    const validationRules = [
-      {
-        required: true,
-        message: t('enterUsername'),
-        whitespace: true
-      },
-      {
-        validator: (rule, value) => {
-          const minLength = 6;
-          return value && value.trim().length < minLength
-            ? Promise.reject(new Error(t('usernameIsTooShort', { length: minLength })))
-            : Promise.resolve();
-        }
-      },
-      {
-        validator: (rule, value) => {
-          return value && forbiddenUsernames.includes(value.toLowerCase())
-            ? Promise.reject(new Error(t('usernameIsInUse')))
-            : Promise.resolve();
-        }
+  const validationRules = [
+    {
+      required: true,
+      message: t('enterUsername'),
+      whitespace: true
+    },
+    {
+      validator: (rule, value) => {
+        const minLength = 6;
+        return value && value.trim().length < minLength
+          ? Promise.reject(new Error(t('usernameIsTooShort', { length: minLength })))
+          : Promise.resolve();
+      }
+    },
+    {
+      validator: (rule, value) => {
+        return value && forbiddenUsernames.includes(value.toLowerCase())
```

Comment:
Also not related to the ticket, but usernames are case-sensitive, we are too strict here. But maybe that's a good thing, what do you think? I don't think we should change it in this PR, just to keep it in mind.

Reply:
Interesting, that's true, usernames are case sensitive. It might be a good thing to keep this restriction here though, so that only the person that has selected a certain username can use the variations of it (given by changing character casings) when they would want to change their username.

It also avoids the confusion that would come with more users having the same username with different casing 🤔

Reply:
Yep, agree.